src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/NocRespondentHelper.java

```diff
@@ -98,24 +98,15 @@ public RespondentSumType getRespondent(String respName, CaseData caseData) {
                 .orElse(new RespondentSumType());
     }
 
-    public Optional<RespondentSumTypeItem> getRespondentSumTypeItem(CaseData caseData,
-                                                                    RepresentedTypeRItem respondentRep) {
-        final List<RespondentSumTypeItem> respondentCollection = caseData.getRespondentCollection();
-        return respondentCollection.stream()
-                .filter(resp ->
-                        resp.getValue().getRespondentName()
-                                .equals(respondentRep.getValue().getRespRepName())).findFirst();
-    }
-
-    public List<RepresentedTypeRItem> updateWithRespondentIds(CaseData caseData) {
-        List<RepresentedTypeRItem> repList = new ArrayList<>();
+    public void updateWithRespondentIds(CaseData caseData) {
```

Comment:
Null checks are needed on this throughout. Also, can you avoid nested for loops somehow?

Reply:
null check added, we can avoid nested loops by writing a stream.filter() which was already there, but it was changing the order of items in the collection in case the value doesn't exist. This way, it will not change the order.